src/components/CmsFilters.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <!-- Filters -->
+  <div v-if="filters.length > 0" class="filters-section row">
+    <div class="full-width container pb-1">
+      <div class="categories-container">
+        <div v-for="category in filters" :key="category" class="categories">
+          <div
+            :style="setCategoryStyle(category)"
+            class="category-pill"
+            @click="selectCategory(category)"
+          >
+            <div class="category-pill-text">
+              {{ category }}
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+
+import cmsClient from '../cmsHttp';
+
+@Component({
+  name: 'cms-filters',
+})
+export default class CmsFilters extends Vue {
+  @Prop(String) public zoneId!: string;
+
+  public filters: string[] = [];
+  public showFilters: boolean = false;
+  public selectedCategory: string | null = null;
+
+  public selectCategory(category: string) {
+    if (this.selectedCategory === category) {
+      this.selectedCategory = null;
+    } else {
+      this.selectedCategory = category;
+    }
+    this.$emit('updateSelectedCategory', this.selectedCategory);
+  }
+
+  private created() {
+    this.loadFilters();
+  }
+
+  private async loadFilters(): Promise<void> {
+    let response;
+    try {
+      response = await cmsClient.fetchFilterCategories(this.zoneId);
+      this.filters = ['All', ...this.shuffleArray<string>(response.data)];
+    } catch (error) {}
+  }
+
+  private setCategoryStyle(category: string): string {
+    if (
+      category === this.selectedCategory ||
+      (this.selectedCategory === null && category === 'All')
+    ) {
+      return 'color: #fff; font-weight: bold; background-color: #1c7bbd;';
+    }
+    return 'background-color: #EFEFEF;';
+  }
+
+  private shuffleArray<T>(array: Array<T>): Array<T> {
```

Comment:
Might be worth an explanation of why we're shuffling

Reply:
Also, if this is a util others would use, probably worth moving it to some kind of shared utils file, otherwise, I think it would be fine to specify String arrays instead of using the generic type here.